horace_core/algorithms/dnd.m

```diff
@@ -0,0 +1,64 @@
+function wout = dnd (win,varargin)
+% Convert input sqw or dnd object, array of sqw/dnd  objects or cellarray
+% of sqw/dnd objects
+% into corresponding d0d, d1d,...d4d object(s)
+%
+%   >> wout = dnd (win)
+%   >> wout = dnd (win,''-cell_return')
+%
+%  If the inputs are convertable to the same shape of dnd objects (e.g. all
+%  d1d or all d3d), the function returns the array of extracted objects
+%  If the inputs correspond to the mixture of dnd objects (e.g. d1d and d2d),
+%  the result is the cellarray containing these objects
+%
+% if '-cell_return' option is provided, the method returns cellarray of dnd
+% objects in any case
```

Comment:
This function name looks like a constructor - ok, actual classes are d1d etc, but would 'convert_to_dnd' be more informative? For dnd inputs, is this not a null operation? Some explanation of what this does for d1-4d's would be useful.